config.js

```diff
@@ -228,216 +217,159 @@ exports.updateFromArgv = function() {
-    }
-    exports.bot_command = argv._;
     exports.check_rejectnew = function()
     {
-        if (argv.rejectnew)  return true;
-        if (argv.rejectnewfile && fs.existsSync(argv.rejectnewfile))  return true;
+        if (optimist.argv.rejectnew)  return true;
+        if (optimist.argv.rejectnewfile && fs.existsSync(optimist.argv.rejectnewfile))  return true;
         return false;
     };
 
-    /* 2) specifc r_u cases :*/
-    if (argv.minrank && !argv.minrankranked && !argv.minrankunranked) {
-        exports.minrank = parseRank(argv.minrank);
-    }
-    if (argv.minrankranked) {
-        exports.minrankranked = parseRank(argv.minrankranked);
-    }
-    if (argv.minrankunranked) {
-        exports.minrankunranked = parseRank(argv.minrankunranked);
-    }
-    if (argv.maxrank && !argv.maxrankranked && !argv.maxrankunranked) {
-        exports.maxrank = parseRank(argv.maxrank);
-    }
-    if (argv.maxrankranked) {
-        exports.maxrankranked = parseRank(argv.maxrankranked);
-    }
-    if (argv.maxrankunranked) {
-        exports.maxrankunranked = parseRank(argv.maxrankunranked);
-    }
-
-    if (argv.bans) {
-        for (let user of argv.bans.split(',')) {
-            exports.banned_users[user] = true;
-        }
-    }
-    if (argv.bansranked) {
-        for (let user of argv.bansranked.split(',')) {
-            exports.banned_users_ranked[user] = true;
-        }
-    }
-    if (argv.bansunranked) {
-        for (let user of argv.bansunranked.split(',')) {
-            exports.banned_users_unranked[user] = true;
+    // r_u exports
+    /* 1) general r_u cases:*/
+    for (let familyNameString of genericMain_r_u_Families) {
+        let argObject = argObject_r_u(optimist.argv, familyNameString);
+        for (let r_u in argObject) {
+            exports[r_u][familyNameString] = argObject[r_u] || false;
         }
     }
 
-    if (argv.boardsizes) {
-        for (let boardsize of argv.boardsizes.split(',')) {
-            if (boardsize === "all") {
-                exports.allow_all_boardsizes = true;
-            } else if (boardsize === "custom") {
-                exports.allow_custom_boardsizes = true;
-                for (let boardsizewidth of argv.boardsizewidths.split(',')) {
-                    exports.allowed_custom_boardsizewidths[boardsizewidth] = true;
-                }
-                for (let boardsizeheight of argv.boardsizeheights.split(',')) {
-                    exports.allowed_custom_boardsizeheights[boardsizeheight] = true;
-                }
-            } else {
-                exports.allowed_boardsizes[boardsize] = true;
-            }
+    /* 2) specifc r_u cases:*/
+    for (let familyNameString of rank_r_u_Families) {
+        let argObject = argObject_r_u(optimist.argv, familyNameString);
+        for (let r_u in argObject) {
+            argObject[r_u] = parseRank(argObject[r_u]);
+            exports[r_u][familyNameString] = argObject[r_u] || false;
```

Comment:
Why overwrite `argObject[r_u]`? Why not 1) make a new variable or 2)"inline parseRank on line 307?

Reply:
i simplified the code with your advice in mind, now we directly export the parsed value, we dont store it inbetween